chips/stm32f429zi/src/ethernet/ethernet.rs

```diff
@@ -0,0 +1,2288 @@
+    }
+
+    fn disable_receive_buffer_unavailable(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::RBUIE::CLEAR);
+    }
+
+    fn is_receive_buffer_unavailable(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::RBUIE)
+    }
+
+    fn enable_underflow_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TUIE::SET);
+    }
+
+    fn disable_underflow_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TUIE::CLEAR);
+    }
+
+    fn is_underflow_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::TUIE)
+    }
+
+    fn enable_overflow_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::ROIE::SET);
+    }
+
+    fn disable_overflow_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::ROIE::CLEAR);
+    }
+
+    fn is_overflow_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::ROIE)
+    }
+
+    fn enable_transmit_jabber_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TJTIE::SET);
+    }
+
+    fn disable_transmit_jabber_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TJTIE::CLEAR);
+    }
+
+    fn is_transmit_jabber_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::TJTIE)
+    }
+
+    fn enable_transmit_process_stopped_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TPSIE::SET);
+    }
+
+    fn disable_transmit_process_stopped_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TPSIE::CLEAR);
+    }
+
+    fn is_transmit_process_stopped(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::TPSIE)
+    }
+
+    fn enable_transmit_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TIE::SET);
+    }
+
+    fn disable_transmit_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TIE::CLEAR);
+    }
+
+    fn is_transmit_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::TIE)
+    }
+
+    fn enable_receive_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::RIE::SET);
+    }
+
+    fn disable_receive_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::RIE::CLEAR);
+    }
+
+    fn is_receive_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::RIE)
+    }
+
+    fn enable_transmit_buffer_unavailable_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TBUIE::SET);
+    }
+
+    fn disable_transmit_buffer_unavailable_interrupt(&self) {
+        self.dma_registers.dmaier.modify(DMAIER::TBUIE::CLEAR);
+    }
+
+    fn is_transmit_buffer_unavailable_interrupt_enabled(&self) -> bool {
+        self.dma_registers.dmaier.is_set(DMAIER::TBUIE)
+    }
+
+    #[allow(dead_code)]
+    fn get_current_host_transmit_descriptor_address(&self) -> u32 {
+        self.dma_registers.dmachtdr.get()
+    }
```

Comment:
I think it may be better to not keep such unused functions around, assuming you don't plan on integrating them in the near term. If required, we can always re-add them later.